meson.build

```diff
@@ -0,0 +1,294 @@
+
+# Define strdup() in string.h under glibc >= 2.10 (POSIX.1-2008)
+add_project_arguments('-D_POSIX_C_SOURCE=200809L', language: 'c')
+
+###############################################################################
+# Dependencies
+
+libmath = cc.find_library('m', required: false)
+
+json = dependency('json-c')
+
+# glib
+gobject = dependency('gobject-2.0', required: get_option('glib'))
+use_glib = gobject.found()
+conf.set10('MYPAINT_CONFIG_USE_GLIB', use_glib)
+
+# GEGL
+gegl = dependency('gegl-0.4', 'gegl-0.3', required: get_option('gegl'))
+use_gegl = gegl.found()
+if use_gegl
+  gegl_gir = gegl.version().version_compare('>=0.4') ? 'Gegl-0.4' : 'Gegl-0.3'
+endif
+
+introspection_required_version = '1.32.0'
+
+
+# OpenMP
+openmp = dependency('openmp', required: get_option('openmp'))
+
+## gperftools ##
+libprofiler = dependency('libprofiler', required: get_option('gperftools'))
+
+# Profiling
+if get_option('profiling')
+  add_project_arguments('-pg', language: 'c')
+endif
+
+# Internationalization
+have_i18n = false
+if get_option('i18n')
+  if not cc.has_header('libintl.h')
+    error('libintl.h must be available for i18n support')
+  endif
+  if platform_win32 or platform_osx
+    libintl = cc.find_library('intl')
+  else
+    libintl = []
+  endif
+  have_i18n = true
+endif
+conf.set10('HAVE_GETTEXT', have_i18n)
+
+
+# Docs
+enable_docs = get_option('docs')
+if enable_docs
+  doxygen = find_program('doxygen')
+  sphinx_build = find_program(
+    'sphinx-build3',
+    'sphinx-build-3',
+    'sphinx-build2',
+    'sphinx-build-2',
+    'sphinx-build',
+  )
+  # todo: the python 'breathe' extension is also a dependency to doc building.
+  # the configure script should check for its existence.
+endif
+
+
+###############################################################################
+# Configure files
+
+toplevel_inc = include_directories('.')
+
+configure_file(
+  output: 'config.h',
+  configuration: conf,
+)
+
+brush_settings_headers = custom_target(
+  'brush_settings_headers',
+  input: 'brushsettings.json',
+  output: [
+    'mypaint-brush-settings-gen.h',
+    'brushsettings-gen.h',
+  ],
+  command: [
+    find_program('python3'),
+    meson.current_source_dir() / 'generate.py',
+    '@OUTPUT@',
+  ],
+  depend_files: [
+    'generate.py',
+  ],
+  install: true,
+  install_dir: [
+    get_option('includedir') / api_name,
+    false,
+  ],
+)
+
+
+###############################################################################
+# Source files
+
+libmypaint_sources = [
+  'brushmodes.c',
+  'fifo.c',
+  'helpers.c',
+  'mypaint-brush-settings.c',
+  'mypaint-brush.c',
+  'mypaint-fixed-tiled-surface.c',
+  'mypaint-mapping.c',
+  'mypaint-matrix.c',
+  'mypaint-rectangle.c',
+  'mypaint-surface.c',
+  'mypaint-symmetry.c',
+  'mypaint-tiled-surface.c',
+  'mypaint.c',
+  'operationqueue.c',
+  'rng-double.c',
+  'tilemap.c',
+]
+
+libmypaint_introspectable_headers = [
+  'mypaint-brush.h',
+  'mypaint-brush-settings.h',
```

Comment:
`BrushInput`, `BrushSetting`, `BrushState` not part of GIR for some reason.

Reply:
Looks like they fail to be found, even though we are adding `mypaint-brush-settings-gen.h` to `libmypaint_introspectable_headers`:

```
../mypaint-symmetry.h:23: Error: MyPaint: identifier not found on the first line:
  * MyPaintSymmetryType: Enumeration of different kinds of symmetry
    ^
../mypaint-symmetry.h:43: Error: MyPaint: identifier not found on the first line:
  * MyPaintSymmetryState: Contains the basis for symmetry calculations
    ^
../mypaint-symmetry.h:58: Error: MyPaint: identifier not found on the first line:
  * MyPaintSymmetryData: Contains data used for symmetry calculations
    ^

```